executor/executor.go

```diff
@@ -289,19 +293,27 @@ func (executor *Executor) execute(
 		Containers: []kuber.ContainerResourcesRequirements{
 			{
 				Name: containerName,
-				Limits: kuber.RequestLimit{
-					Memory: decision.ContainerResources.Limits.Memory,
-					CPU:    decision.ContainerResources.Limits.CPU,
-				},
-				Requests: kuber.RequestLimit{
-					Memory: decision.ContainerResources.Requests.Memory,
-					CPU:    decision.ContainerResources.Requests.CPU,
-				},
+				Limits: kuber.RequestLimit{},
+				Requests: kuber.RequestLimit{},
 			},
 		},
 	}
-
-
+	if decision.ContainerResources.Requests != nil {
+		if decision.ContainerResources.Requests.CPU != nil {
+			totalResources.Containers[0].Requests.CPU = decision.ContainerResources.Requests.CPU
+		}
+		if decision.ContainerResources.Requests.Memory != nil {
+			totalResources.Containers[0].Requests.Memory = decision.ContainerResources.Requests.Memory
```

Comment:
You are always assuming that the number of containers is 1, is that because it's a patch request and it will only impact the selected container or will overwrite all the containers?
```
totalResources.Containers[0]
```

@mohamedveron

Reply:
because we execute just one recommendation

Reply:
and it will affect one container